docs/source/recipes/how-to-use-color-picker-widget.md

```diff
@@ -0,0 +1,139 @@
+It can be used in your custom block settings schema or enhance an existing block as follows:
+
+```js hl_lines="13-16, 31-42"
+import { addStyling } from '@plone/volto/helpers/Extensions/withBlockSchemaEnhancer';
+import { defineMessages } from 'react-intl';
+import config from '@plone/volto/registry';
+
+const messages = defineMessages({
+  backgroundColor: {
+    id: 'Background color',
+    defaultMessage: 'Background color',
+  },
+});
+
+export const defaultStylingSchema = ({ schema, formData, intl }) => {
+  const BG_COLORS = [
+    { name: 'transparent', label: 'Transparent' },
+    { name: 'grey', label: 'Grey' },
+  ];
+
+  // You could allow passing the color definition from the config or from the default
+  // defined above
+  const colors =
+    config.blocks?.blocksConfig?.[formData['@type']]?.colors || BG_COLORS;
+
+  // Same for the default used (or undefined)
+  const defaultBGColor =
+    config.blocks?.blocksConfig?.[formData['@type']]?.defaultBGColor;
+
+  // This adds the StylingWrapper support to your block
+  addStyling({ schema, intl });
+
+  // Then we add the field to the fieldset inside the StylingWrapper `styles` field schema fieldset array
+  schema.properties.styles.schema.fieldsets[0].fields = [
+    ...schema.properties.styles.schema.fieldsets[0].fields,
+    'backGroundColor',
+  ];
+
+  // and finally, we add the field to the StylingWrapper `styles` object field schema properties
+  schema.properties.styles.schema.properties['backGroundColor'] = {
+    widget: 'color_picker',
+    title: intl.formatMessage(messages.backgroundColor),
+    colors,
+    default: defaultBGColor,
+  };
+
+  return schema;
+};
+```
+
+The widget name the color picker widget uses is `color_picker`.
+
+## Color definitions
+
+The `colors` property of the widget controls which colors are available to choose in the widget.
+This is the signature of the object:
+
+```ts
+type Color =
+  | {
+      label: string;
+      style: Record<`--${string}`, string> & { name: string };
+    }
+  | {
+      name: string;
+      label: string;
+    };
+
+const colors: Color[] = [
+  { style: { name: 'red', '--background-color': 'red' }, label: 'red' },
+  {
+    style: { name: 'yellow', '--background-color': 'yellow' },
+    label: 'yellow',
+  },
+  { name: 'green', label: 'green' },
+]
+```
+
+## Using basic definitions
+
+The basic color definition is the one that saves a string label as the widget value.
+You can use it on your own code by reading it from the resultant data and use it according your designed solution.
+
+In the case of using it along with the StyleWrapper, the value will be injected as a class name of the form:
+
+```html
+<div class="has--backgroundColor--ee22ee">
+  ...
+</div>
+```
+
+Then you should create the CSS rules according to these injected class names.
+
+## Using custom CSS properties as color definitions
+
+The `style` key defines a set of custom CSS properties to be saved as widget value.
```

Comment:
I'm not sure what you intend here with "to be saved as widget value", but I think you meant this.

```suggestion
The `style` key defines a set of custom CSS properties to be added as the value to the HTML attribute `style`.
```